mono/metadata/locales.c

```diff
@@ -740,7 +740,8 @@ int ves_icall_System_Globalization_CompareInfo_internal_compare (MonoCompareInfo
 					 options));
 }
 
-void ves_icall_System_Globalization_CompareInfo_assign_sortkey (MonoCompareInfo *this_obj, MonoSortKey *key, MonoString *source, gint32 options)
+MonoArray*
+ves_icall_System_Globalization_CompareInfo_assign_sortkey (MonoString *source)
 {
 	ERROR_DECL (error);
```

Comment:
This doesn't seem to do anything useful anymore, it converts a string to its bytes.

Reply:
Yeah you are right -- just an identity. Just rewrite it in all managed?

There are a few confusing things.
In particular, I thought I saw a similar crash in an unrelated build.
Like maybe monolite got updated with the change, and maybe we depend on the data being corrupted?
Maybe pretend I didn't see anything confusing and proceed.

Removing the icall and just doing it in managed seems reasonable. Let me try that.

Reply:
Identity, but truncating guint16 to guint8, granted.

Reply:
Ugh.
```
ReferenceSources/CompareInfo.cs(85,8): error CS0122: 'SortKey.key' is inaccessible due to its protection level
make[8]: *** [../../class/lib/build-macos/mscorlib.dll] Error 1
```